router/pkg/config/config.go

```diff
@@ -577,6 +577,19 @@ type WebSocketConfiguration struct {
 	ForwardInitialPayload bool `yaml:"forward_initial_payload" envDefault:"true" env:"WEBSOCKETS_FORWARD_INITIAL_PAYLOAD"`
 	// Authentication configuration for the WebSocket Connection
 	Authentication WebSocketAuthenticationConfiguration `yaml:"authentication,omitempty"`
+	// SetClientInfoFromInitialPayload configuration for the WebSocket Connection
+	SetClientInfoFromInitialPayload WebSocketSetClientInfoFromInitialPayloadConfiguration `yaml:"set_client_info_from_initial_payload"`
+}
+
+type WebSocketSetClientInfoFromInitialPayloadConfiguration struct {
+	// Enabled true if the Router should set the client info from the initial payload of a Subscription Request to the Subgraph
+	Enabled bool `yaml:"enabled" envDefault:"true" env:"WEBSOCKETS_SET_CLIENT_INFO_FROM_INITIAL_PAYLOAD_ENABLED"`
+	// ExportToClientRequestHeaders true if the Router should export the client info to the client request headers
+	ExportToClientRequestHeaders bool `yaml:"export_to_client_request_headers" envDefault:"true" env:"WEBSOCKETS_SET_CLIENT_INFO_FROM_INITIAL_PAYLOAD_EXPORT_TO_CLIENT_REQUEST_HEADERS"`
+	// ClientNameHeaderName is the name of the header where the client name should be exported to
+	ClientNameHeaderName string `yaml:"client_name_header_name" envDefault:"graphql-client-name" env:"WEBSOCKETS_SET_CLIENT_INFO_FROM_INITIAL_PAYLOAD_CLIENT_NAME_HEADER_NAME"`
```

Comment:
I suggest to make it an array because in the wild we already support apollo and `graphql-client-name` convention

Reply:
We export the value into a header, that makes no sense. 😅

Reply:
I took me a moment to understand you abstraction. Why do we want to make this fix? Let the user define which property will be used as value.

```yaml
enabled: true
source_field: "graphqlClientName"
target_header: "my-client-header"
```

`ExportToClientRequestHeaders` is not needed, you can check for existence of `target_header`